modules/candidate_parameters/ajax/formHandler.php

```diff
@@ -363,71 +363,68 @@ function editConsentStatusFields($db, $user)
+    $pscid = $candidate->getPSCID();
 
-    $consent_details =\Utility::asArray($consent['Consent']);
-    if (!$consent_details[0]) {
-        // If only one consent, need to put in an array
-        $temp            = array();
-        $temp[]          = $consent_details;
-        $consent_details = $temp;
-    }
-
-    foreach ($consent_details as $consentType) {
+    // Get list of all consent types
+    $consentDetails = \Utility::getConsentList();
+    // Get list of consents for candidate
+    $candidateConsent = $candidate->getConsents();
 
-        $consentName       = $_POST[$consentType['name']];
-        $consentDate       = $_POST[$consentType['name'] . '_date'];
-        $consentWithdrawal = $_POST[$consentType['name'] . '_withdrawal'];
+    foreach ($consentDetails as $consentID=>$consent) {
+        $consentName  = $consent['Name'];
+        $consentLabel = $consent['Label'];
 
         // Process posted data
-        $consent    = (isset($consentName) && $consentName !== "null") ?
-            $consentName : null;
-        $date       = (isset($consentDate) && $consentDate !== "null") ?
-            $consentDate : null;
-        $withdrawal = (isset($consentWithdrawal) && $consentWithdrawal !== "null") ?
-            $consentWithdrawal : null;
-
-        $updateValues = [
-                         'CandID'                               => $candID,
-                         'entry_staff'                          => $id,
-                         $consentType['name']                   => $consent,
-                         ($consentType['name'] . '_date')       => $date,
-                         ($consentType['name'] . '_withdrawal') => $withdrawal,
-                        ];
-
-        $newRecord = true;
-
-        if ($candID) {
-            $exists = $db->pselectOne(
-                "SELECT * from participant_status WHERE CandID=:candid",
-                ['candid' => $candID]
-            );
-            if ($exists && count($exists) > 0) {
-                $newRecord = false;
-            }
-
-            if ($newRecord) {
-                $db->insert('participant_status', $updateValues);
-            } else {
+        $status     = ($_POST[$consentName] !== 'null') ?
+                        $_POST[$consentName] : null;
+        $date       = ($_POST[$consentName . '_date'] !== 'null') ?
+                        $_POST[$consentName . '_date'] : null;
+        $withdrawal = ($_POST[$consentName . '_withdrawal'] !== 'null') ?
+                        $_POST[$consentName . '_withdrawal'] : null;
+
+        $updateStatus  = [
+                          'CandidateID'   => $candID,
+                          'ConsentID'     => $consentID,
+                          'Status'        => $status,
+                          'DateGiven'     => $date,
+                          'DateWithdrawn' => $withdrawal,
+                         ];
+        $updateHistory = [
+                          'PSCID'         => $pscid,
```

Comment:
If I'm reading this code correctly, the rel table is updated with candidate ID, but this history table is updated with PSCID? Why the distinction?

Reply:
CandID for back-end, PSCID for front-end. the history is only really used to keep a log and show it on the front-end

Reply:
gotcha